_posts/2025-07-25-model-registry-integration.md

```diff
@@ -0,0 +1,445 @@
+
+    echo ""
+    echo "🎉 Installation complete! 🎉"
+    echo ""
+    echo "📋 Next steps:"
+    echo "   1. Run this command to start port forwarding:"
+    echo "      kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80"
+    echo ""
+    echo "   2. Open your browser and go to:"
+    echo "      http://localhost:8080"
+    echo ""
+    echo "   3. Login with these credentials:"
+    echo "      Username: user@example.com"
+    echo "      Password: 12341234"
+    echo ""
+```
+
+**Note:** This script assumes `kustomize` and `jq` are installed and in your PATH. Ensure you have followed the prerequisites section to install these tools. Also, ensure you have cloned the `kubeflow/manifests` repository to `~/kubeflow-manifests-repo` and are running this script from that directory.
+
+
+
+**Accessing the cluster:**
+ After this complex setup, you would typically access the Kubeflow Central Dashboard via an Ingress or LoadBalancer, as described in the [Connect to your Kubeflow cluster](https://github.com/kubeflow/manifests?tab=readme-ov-file#connect-to-your-kubeflow-cluster) section of the `kubeflow/manifests` README.
+
+***Once the full multi-user Kubeflow is deployed, you would then apply the Model Registry UI's Istio-dependent overlay:***
+```
+cd ~/kubeflow-manifests-repo/applications/model-registry/upstream/options/ui/overlays/istio
+kubectl apply -k . -n kubeflow
+```
+
+
+This would then work because Istio would be present.
+
+- Integrate Model Registry UI with Kubeflow Central Dashboard (Optional but Recommended): This step adds a direct link to the Model Registry UI in the Kubeflow Central Dashboard's navigation menu.
+```
+kubectl get configmap centraldashboard-config -n kubeflow -o json | \
+jq '.data.links |= (fromjson | .menuLinks += [{"icon": "assignment", "link": "/model-registry/", "text": "Model Registry", "type": "item"}] | tojson)' | \
+kubectl apply -f - -n kubeflow
+```
+
+***Note: This command requires jq to be installed (sudo dnf install jq). If you prefer, you can edit the ConfigMap manually: kubectl edit configmap -n kubeflow centraldashboard-config and add the menu item under data.links.menuLinks.***
+
+**Set Up Port-Forwards:** 
+
+Open three separate terminal windows and run:
+
+- KFP UI: `kubectl port-forward -n kubeflow svc/ml-pipeline-ui 8080:80`
+- Model Registry API: `kubectl port-forward -n kubeflow svc/model-registry-service 8082:8080`
+- Model Registry UI: `kubectl port-forward -n kubeflow svc/model-registry-ui-service 8084:80`
+- Minio (for KFP logs): `kubectl; port-forward -n kubeflow svc/minio-service 9000:9000`
+
+# Connecting KFP to Model Registry: Best Practices
+
+To enable your KFP component to register models, you need to provide it with the Model Registry's API endpoint and, ideally, an authentication token. While the Model Registry API can be accessed directly, using a Kubernetes Secret for sensitive information like tokens is a best practice.
```

Comment:
When using the install.sh script, it seemed the Model Registry API didn't require any authentication or authorization. Did you add this after the fact?